### DISCUSSION TITLE
new data isn't loaded consistently on refresh

### DISCUSSION THREAD

**Discussion by saarwasserman** [kiali/kiali#4444] [Q&A]
my kiali 1.34.1 setup works well overall however when refresh takes place the new data is not showing up sometimes. after some manual or some next automatic refresh happens it does. what can be the cause? (same prometheus setup seems to work fine with other tools.)

**Comment by lucasponce** [comment]
Hi @saarwasserman, 
I'd need more information but if you are talking about seeing new namespaces in Kiali, those are cached (you can configure through the Kiali CR the refresh rate).

As Kiali cannot run with cluster permissions, it can't create a informer to LIST+WATCH events and react when a new namespace is happening, so typically you'd see the new namespaces when cache expires.

About to refresh the telemetry data, it also depends of the scrape ratio of your prometheus configuration, in other words, if prometheus query the telemetry every 5 minutes, Kiali won't process new info in shorter periods.

To adjust this, you'd need to configure your prometheus server.

Other elements, like workloads (controllers + pods), services, or other configuration resources have proper informers, so, information is cached locally but updated when a new event is captured, so the "refresh" should take seconds for these elements, so I guess you don't refer to them.

Hope this helps.

**Reply by saarwasserman** [reply]
The exact scenario is that the graphs sometimes shows 'empty graph' after refresh, however when I manual refresh a second later it shows a full graph. I set the interval to be 5 min so it should contain some traffic at that period.

**Reply by jmazzitelli** [reply]
Look in your kiali pod's logs to see if you get any error messages. Perhaps the empty graph is caused by some error that occurs in the backend that causes the graph not to render (??)

**Reply by saarwasserman** [reply]
I don't see any related error in the backend logs. I see in browser's console that when the graph is not rendered there is an uncaught promise which indicates {'isCanceled': true}, from time to time. followed 1min after by the message about mice sensitivity warning. although I guess they are not related.

**Reply by saarwasserman** [reply]
wondering is it's data not retrieved from prom, not enough resources for kiali or frontend issue.

**Reply by jshaughn** [reply]
It doesn't sound like a front-end issue.  It seems more like the backend graph requests are returning successfully, but with no data.  "Empty Graph" is different than if there were an error.  If you were to examine the Network requests sent from the browser (F12, Network tab, filter by 'graph'), I would expect you'd see it come back with a 200, but just a small JSON with no nodes or edges.  So then the question is why would the backend return nothing.  There was a rare case a while back where prometheus was actually being load balanced and some requests ended up going to an empty prom.  Maybe check the prometheus logs and make sure requests are not getting rejected due to some sort of pressure limiting.

**Reply by saarwasserman** [reply]
thanks for your responses. I'll take a look at prometheus.
